ArraySimulation/MPPT/GlobalMPPTAlgo/GlobalAlgorithm.py

```diff
@@ -0,0 +1,127 @@
+            algorithm.
+        MpptAlgoType: String
+            The name of the model type.
+        strideType: String
+            The name of the stride model type.
+        """
+        self.MAX_VOLTAGE = self.MAX_VOLTAGE_PER_CELL*numCells
+        self._MpptAlgoType = MpptAlgoType
+        
+        
+        if MpptAlgoType == "PandO":
+            self._model = PandO(numCells, strideType)
+        elif MpptAlgoType == "IC":
+            self._model = IC(numCells, strideType)
+        elif MpptAlgoType == "Ternary":
+            self._model = Ternary(numCells, strideType)
+        elif MpptAlgoType == "Golden":
+            self._model = Golden(numCells, strideType)
+        elif MpptAlgoType == "IC":
+            self._model = IC(numCells, strideType)
+        elif MpptAlgoType == "Bisection":
+            self._model = Bisection(numCells, strideType)
+        elif MpptAlgoType == "FC":
+            self._model = FC(numCells, strideType)
+            print("Hello")
+        elif MpptAlgoType == "Default":
+            self._model = MPPTAlgorithm(numCells, MpptAlgoType, strideType)
+        else:
+            self._model = MPPTAlgorithm(numCells, MpptAlgoType, strideType)
+        
+        self.vOld = 0.0
+        self.iOld = 0.0
+        self.tOld = 0.0
+        self.irrOld = 0.0
+        self.pOld = 0.0
+    def getReferenceVoltage(self, arrVoltage, arrCurrent, irradiance, temperature):
+        """
+        Calculates the reference voltage output for the given PVSource output.
+        May use prior history.
+
+        Parameters
+        ----------
+        arrVoltage: float
+            Array voltage in V.
+        arrCurrent: float
+            Array current in A.
+        irradiance: float
+            Irradiance in W/M^2 (G)
+        temperature: float
+            Cell Temperature in C.
+
+        Return
+        ------
+        float The reference voltage that should be applied to the array in the
+        next cycle.
+
+        Assumptions
+        -----------
+        This method is called sequentially in increasing cycle order. The
+        arrVoltage and arrCurrent are expected to have stabilized to the
+        reference voltage applied in the last cycle, if any.
+
+        Note that the second assumption doesn't hold true in reality, as large
+        changes in reference voltage may mean the array does not converge to
+        steady state behavior by the next MPPT cycle. This should always be
+        considered in the algorithms.
+        """
+        return self._model.getReferenceVoltage(arrVoltage,arrCurrent,irradiance,temperature)
+
+    def reset(self):
+        """
+        Resets any internal variables set by the MPPT algorithm during operation.
+        """
+        self._model.reset()
+        self.vOld = 0.0
+        self.iOld = 0.0
+        self.pOld = 0.0
+        self.irrOld = 0.0
+        self.tOld = 0.0
+
+    def getMPPTType(self):
```

Comment:
should be getMPPTGlobalType, not to be confused with MPPTAlgorithm's getMPPTType, which should be clarified to getMPPTLocalType.